base/compiler/abstractinterpretation.jl

```diff
@@ -2291,33 +2291,14 @@ function abstract_eval_statement_expr(interp::AbstractInterpreter, e::Expr, vtyp
         t = rt
     elseif ehead === :new
         t, isexact = instanceof_tfunc(abstract_eval_value(interp, e.args[1], vtypes, sv))
-        nothrow = true
-        if isconcretedispatch(t)
-            ismutable = ismutabletype(t)
-            fcount = fieldcount(t)
+        ut = unwrap_unionall(t)
```

Comment:
Since we have some tests that ensure `!has_free_typevars(ut)` at runtime. Just double-checking that this test is compatible with the consistent and nothrow here. I think it looks right?
```
3783 @test_throws TypeError("new", DataType, Complex) eval(Expr(:new, Complex))
   1 @test_throws TypeError("new", DataType, Complex.body) eval(Expr(:new, Complex.body))
   2 @test_throws TypeError("new", DataType, Complex) eval(Expr(:splatnew, Complex, ()))
   3 @test_throws TypeError("new", DataType, Complex.body) eval(Expr(:splatnew, Complex.body, ()))
```

Reply:
This doesn't change our definition of nothrow, only consistent, so I think those tests should be unaffected.